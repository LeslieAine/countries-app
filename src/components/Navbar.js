import { NavLink, Outlet, useLocation } from 'react-router-dom';
import { IoChevronBack } from 'react-icons/io5';
import styles from '../styles/Navbar.module.css';

const Navbar = () => {
  const location = useLocation();

  return (
    <div className={styles.navbar}>
      <NavLink to="/" className={styles.link}>
        { location.pathname !== '/' && <IoChevronBack className={styles.back} /> }
      </NavLink>
      <Outlet />
    </div>
  );
};

export default Navbar;
