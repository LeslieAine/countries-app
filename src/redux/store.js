import countriesReducer from './countries/countriesSlice';

const { configureStore } = require('@reduxjs/toolkit');

const store = configureStore({
  reducer: {
    countries: countriesReducer,
  },
});

export default store;
