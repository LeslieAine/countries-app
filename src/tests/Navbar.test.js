import React from 'react';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import store from '../redux/store';
import Navbar from '../components/Navbar';

describe('Navbar', () => {
  it('should render the component', () => {
    const tree = renderer.create(
      <Provider store={store}>
        <BrowserRouter>
          <Navbar />
        </BrowserRouter>
      </Provider>,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
