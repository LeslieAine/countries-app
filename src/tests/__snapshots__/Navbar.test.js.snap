// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar should render the component 1`] = `
<div
  className="navbar"
>
  <a
    aria-current="page"
    className="link active"
    href="/"
    onClick={[Function]}
  />
</div>
`;
